streamingvisitors/src/vespa/vsm/vsm/docsumfilter.cpp

```diff
@@ -403,28 +394,23 @@ DocsumFilter::get_flattened_summary_field(const DocsumFieldSpec& field_spec, con
     return DocsumStoreFieldValue(std::move(value));
 }
 
-search::docsummary::DocsumStoreFieldValue
-DocsumFilter::get_summary_field(uint32_t entry_idx, const Document& doc)
+void
+DocsumFilter::insert_summary_field(uint32_t entry_idx, const Document& doc, vespalib::slime::Inserter& inserter)
 {
     const auto& field_spec = _fields[entry_idx];
-    if (field_spec.is_struct_or_multivalue()) {
-        return get_struct_or_multivalue_summary_field(field_spec, doc);
-    } else {
-        if (field_spec.getInputFields().size() == 1 && field_spec.getCommand() == VsmsummaryConfig::Fieldmap::Command::NONE) {
-            const DocsumFieldSpec::FieldIdentifier & fieldId = field_spec.getInputFields()[0];
-            const document::FieldValue* field_value = doc.getField(fieldId.getId());
-            return DocsumStoreFieldValue(field_value);
-        } else if (field_spec.getInputFields().empty() && field_spec.getCommand() == VsmsummaryConfig::Fieldmap::Command::NONE) {
-            return {};
-        } else {
-            return get_flattened_summary_field(field_spec, doc);
+    auto single_source_field_id = get_single_source_field_id(field_spec);
+    if (single_source_field_id.has_value()) {
+        auto field_value = doc.getField(single_source_field_id.value());
+        if (field_value != nullptr) {
+            CheckUndefinedValueVisitor check_undefined;
```

Comment:
Can we add a static function in SlimeFiller for this?

Reply:
Yes. Needs rebase.